src/compiler/checker.ts

```diff
@@ -43172,28 +43181,15 @@ namespace ts {
                     return grammarErrorOnNode(assertionArgument, Diagnostics.Dynamic_import_only_supports_a_second_argument_when_the_module_option_is_set_to_esnext);
                 }
             }
-            if (nodeArguments.length !== 1) {
-                return grammarErrorOnNode(node, Diagnostics.Dynamic_import_must_only_have_a_specifier_and_an_optional_assertion_as_arguments);
-            }
-            return false;
-        }
-
-        function checkGrammarImportCallExpression(node: ImportCall): boolean {
-            if (moduleKind === ModuleKind.ES2015) {
-                return grammarErrorOnNode(node, Diagnostics.Dynamic_imports_are_only_supported_when_the_module_flag_is_set_to_es2020_esnext_commonjs_amd_system_or_umd);
-            }
 
-            if (node.typeArguments) {
-                return grammarErrorOnNode(node, Diagnostics.Dynamic_import_cannot_have_type_arguments);
+            if (nodeArguments.length  === 0 || nodeArguments.length > 2) {
+                return grammarErrorOnNode(node, Diagnostics.Dynamic_import_must_only_have_a_specifier_and_an_optional_assertion_as_arguments);
             }
 
-            const nodeArguments = node.arguments;
-            if (checkGrammarImportCallArguments(node, nodeArguments)) {
-                return true;
-            }
             // see: parseArgumentOrArrayLiteralElement...we use this function which parse arguments of callExpression to parse specifier for dynamic import.
             // parseArgumentOrArrayLiteralElement allows spread element to be in an argument list which is not allowed as specifier in dynamic import.
-            if (nodeArguments.length && isSpreadElement(nodeArguments[0])) {
+            const spreadElement = forEach(nodeArguments, isSpreadElement);
+            if (spreadElement) {
                 return grammarErrorOnNode(nodeArguments[0], Diagnostics.Specifier_of_dynamic_import_cannot_be_spread_element);
             }

```

Comment:
Hmm. This doesn't quite do what we want. We want to report the error on the spread element, not the first argument (which might not be the spread element). I think we want this instead:

```suggestion
            const spreadElement = find(nodeArguments, isSpreadElement);
            if (spreadElement) {
                return grammarErrorOnNode(spreadElement, Diagnostics.Specifier_of_dynamic_import_cannot_be_spread_element);
            }

```